packages/syntax/src/highlight.d.ts

```diff
@@ -0,0 +1,214 @@
+interface TokenObject {
+  kind: string;
+  children: Array<string | TokenObject>;
+}
+
+interface RootNode {
+  children: Array<TokenObject | string>;
+}
+
+interface LanguageDefinition {
+  aliases?: Array<string>;
+  keywords?: string | Record<string, string>;
+  disableAutodetect?: boolean;
+  illegal?: string;
+  contains?: Array<any>;
+  exports?: {
+    preprocessor: any;
+    strings: any; // Maybe Array<string>
+    keywords: any; // Probably the same as keywords above
+  };
+}
+
+declare class TokenTree {
+  constructor();
+  top: () => Array<any>;
+  root: () => RootNode;
+  rootNode: RootNode;
+  stack: Array<RootNode>;
+  add(node: TokenObject | string): void;
+  openNode(kind: string): void;
+  closeNode(): void;
+  closeAllNodes(): void;
+  toJSON(): string;
+  walk<T extends any>(builder: T): T;
+  static _walk<T extends any>(
```

Comment:
`T extends any` is equivalent to `T` afaik

Reply:
Ahh yep I believe you're right.